receiver/nginxreceiver/integration_test.go

```diff
@@ -92,24 +93,39 @@ func (suite *NginxIntegrationSuite) TestNginxScraperHappyPath() {
 	ilm := ilms.At(0)
 	ms := ilm.Metrics()
 
-	require.Equal(t, 7, ms.Len())
-
-	metricValues := make(map[string]int64, 7)
+	require.Equal(t, 4, ms.Len())
 
 	for i := 0; i < ms.Len(); i++ {
 		m := ms.At(i)
 
-		var dps pdata.IntDataPointSlice
-
-		switch m.DataType() {
-		case pdata.MetricDataTypeIntGauge:
-			dps = m.IntGauge().DataPoints()
-		case pdata.MetricDataTypeIntSum:
-			dps = m.IntSum().DataPoints()
+		switch m.Name() {
+		case metadata.M.NginxRequests.Name():
+			require.Equal(t, 1, m.IntSum().DataPoints().Len())
+			require.True(t, m.IntSum().IsMonotonic())
+		case metadata.M.NginxConnectionsAccepted.Name():
+			require.Equal(t, 1, m.IntSum().DataPoints().Len())
+			require.True(t, m.IntSum().IsMonotonic())
+		case metadata.M.NginxConnectionsHandled.Name():
+			require.Equal(t, 1, m.IntSum().DataPoints().Len())
+			require.True(t, m.IntSum().IsMonotonic())
+		case metadata.M.NginxConnectionsCurrent.Name():
+			dps := m.IntGauge().DataPoints()
+			require.Equal(t, 4, dps.Len())
+			for j := 0; j < dps.Len(); j++ {
+				dp := dps.At(j)
+				state, _ := dp.LabelsMap().Get("state")
+				switch state {
+				case "active":
+				case "reading":
+				case "writing":
+				case "waiting":
```

Comment:
Do we expect each data point to be present exactly once? If so, we should likely validate that we see them all. Minimally:
```go
present := map[string]bool {}
...
case "X":
  present["X"] = true
...
require.Equal(t, 7, len(present))
```